### ISSUE TITLE
Bohrium will not build with -Wall -Werror

### ISSUE THREAD

**Issue by tblum** [bh107/bohrium#32]
There are a few warnings @madsbk when building with CMAKE_BUILD_TYPE=Debug.
There are ALOT of warnings when building with CMAKE_BUILD_TYPE=Release. @safl and @tblum and @madsbk 
Could we clean this up? I think we should get to where all build types build with  -Wall -Werror flags set. 

Pretty please with sugar on top :)

Speaking of flags: I thought we moved to -std=c++11? It is still -std=gnu++0x, which is depricated

**Comment by tblum** [comment]
Think I fixed my warnings:
https://github.com/bh107/bohrium/commit/618c56539aa3ceca2189376a408bf1df9c4642e7
https://github.com/bh107/bohrium/commit/8bd67a5b60049c52a42b90f78bfe33dcb3163e7d

**Comment by kenkendk** [comment]
Should this issue be closed then? 

The majority of warnings I see on OSX are deprecation warnings from OpenCL.
Perhaps these should be suppressed.

The only other warnings I see are:
```
bohrium/ve/cpu/bh_ve_cpu.cpp:37:12: warning: unused variable 'TAG' [-Wunused-const-variable]
const char TAG[] = "Component";

bohrium/ve/cpu/utils.cpp:9:12: warning: unused variable 'TAG' [-Wunused-const-variable]
const char TAG[] = "Utils";

bohrium/filter/bcexp/component_interface.cpp:53:17: warning: format specifies type 'long' but the argument has type 'bh_intp' (aka 'long long') [-Wformat]
                myself.nchildren);

bohrium/bridge/c/../cpp/bxx/generator.hpp:137:35: warning: absolute value function 'abs' given an argument of type 'const int64_t' (aka 'const long long') but
      has parameter of type 'int' which may cause truncation of value [-Wabsolute-value]
        nelem = (start-adj_end+1)/abs(skip);
```

**Comment by tblum** [comment]
I see more warnings in bh_dag.h:
´´´
In file included from /home/troels/bohrium/bohrium/fuser/gentle/fuser.cpp:21:0:
/home/troels/bohrium/bohrium/fuser/gentle/../../include/bh_dag.h: In member function ‘bohrium::dag::Vertex bohrium::dag::GraphDW::add_vertex(const bh_ir_kernel&)’:
/home/troels/bohrium/bohrium/fuser/gentle/../../include/bh_dag.h:83:16: warning: unused variable ‘w’ [-Wunused-variable]
         Vertex w = boost::add_vertex(_bglW);
                ^
```
And some different warnings in vem_cluster. I am building with ```CMAKE_BUILD_TYPE=Release``` and 
```
gcc (Ubuntu 4.8.4-2ubuntu1~14.04) 4.8.4
Copyright (C) 2013 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
I think it would be neet if we could set ```-Wall -Werror``` on the master branch. Maybe we need to have Travis CI test with more different gcc versions?

**Comment by madsbk** [comment]
Fixed by #151
